pilot/pkg/xds/internalgen.go

```diff
@@ -150,6 +163,106 @@ func (sg *InternalGen) Generate(proxy *model.Proxy, push *model.PushContext, w *
 			res = append(res, util.MessageToAny(v.xdsNode))
 		}
 		sg.Server.adsClientsMutex.RUnlock()
+	case TypeDebugSyncronization:
+		res = sg.debugSyncz()
+	case TypeDebugConfigDump:
+		if len(w.ResourceNames) == 0 {
+			log.Infof("%s without ResourceName", TypeDebugConfigDump)
+			break
+		}
+		var err error
+		res, err = sg.debugConfigDump(w.ResourceNames[0])
+		if err != nil {
+			log.Infof("%s failed: %v", TypeDebugConfigDump, err)
+			break
+		}
+	}
+	return res
+}
+
+// isSidecar ad-hoc method to see if connection represents a sidecar
+func isSidecar(con *Connection) bool {
```

Comment:
Suggest replacing with `con.node.Type == model.SidecarProxy`. Or rather, `con.node.Type != model.DebugProxy`, since you want to to include ingress? As a result the function name is misleading (includes gateway)